### PR TITLE
Refactor utils/http.go

### LINKED ISSUE #11899: Refactor utils/http.go
Move and refactor the whole `utils/http.go`  to a new package `pkg/requestdecorator`, refactor code accordingly and add unit tests.

### PR DESCRIPTION
**Pull request by runcom**
Fixes: #11899

Signed-off-by: Antonio Murdaca me@runcom.ninja

### CODE REVIEW

File: pkg/requestdecorator/requestdecorator.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
+package requestdecorator
```
Comment (LK4D4) [inline pkg/requestdecorator/requestdecorator.go]: Can you copy `README.md` here as `Package requestdecorator...`. To parse it with godoc

Comment (runcom) [inline pkg/requestdecorator/requestdecorator.go]: sure!

**Comment by runcom** [comment]
done @LK4D4

**Comment by LK4D4** [comment]
LGTM

**Comment by crosbymichael** [comment]
LGTM

**Pull request merged**